metadata-ingestion/src/datahub/ingestion/source/sql/vertica.py

```diff
@@ -18,254 +28,1371 @@
+        properties = table_info.get("properties", {})
+        return description, properties, location
+
+    def is_dataset_eligible_for_profiling(
+        self,
+        dataset_name: str,
+        sql_config: SQLAlchemyConfig,
+        inspector: Inspector,
+        profile_candidates: Optional[List[str]],
+    ) -> bool:
+        """Return whether table, view or projection is  eligible for profiling
+        Args:
+            dataset_name (str): dataset name
+            sql_config (SQLAlchemyConfig): configuration
+            inspector (Inspector): inspector obj from reflection
+            profile_candidates (Optional[List[str]])
+        Returns:
+            bool: returns bool
+        """
+        super().is_dataset_eligible_for_profiling(dataset_name,
+                                                  sql_config, inspector, profile_candidates
+                                                  )
+        return (
 
-def get_view_definition(self, connection, view_name, schema=None, **kw):
-    if schema is not None:
-        schema_condition = "lower(table_schema) = '%(schema)s'" % {
-            "schema": schema.lower()
-        }
-    else:
-        schema_condition = "1"
+            sql_config.projection_pattern.allowed(dataset_name)
+            and sql_config.profile_pattern.allowed(dataset_name)
+
+        )
+
+    def loop_profiler_requests_for_projections(
+        self,
+        inspector: Inspector,
+        schema: str,
+        sql_config: SQLAlchemyConfig,
+    ) -> Iterable["GEProfilerRequest"]:
+        """Function is used for collecting profiling related information for every projections
+            inside an schema.
+
+        Args: schema: schema name
+
+        """
+        from datahub.ingestion.source.ge_data_profiler import GEProfilerRequest
+
+        tables_seen: Set[str] = set()
+        profile_candidates = None  # Default value if profile candidates not available.
+        # if (
+        #     sql_config.profiling.profile_if_updated_since_days is not None
+        #     or sql_config.profiling.profile_table_size_limit is not None
+        #     or sql_config.profiling.profile_table_row_limit is None
+        # ):
+        #     try:
+        #         threshold_time: Optional[datetime.datetime] = None
+        #         if sql_config.profiling.profile_if_updated_since_days is not None:
+        #             threshold_time = datetime.datetime.now(
+        #                 datetime.timezone.utc
+        #             ) - datetime.timedelta(
+        #                 sql_config.profiling.profile_if_updated_since_days
+        #             )
+        #         profile_candidates = self.generate_profile_candidates(
+        #             inspector, threshold_time, schema
+        #         )
+        #     except NotImplementedError:
+        #         logger.debug("Source does not support generating profile candidates.")
+        super().loop_profiler_requests(
```

Comment:
this isn't doing anything - it should be a "yield from"

additionally this method should be overriding loop_profiler_requests instead of being a separate method